plugins/Quotes/class.quotes.plugin.php

```diff
@@ -422,29 +427,21 @@ protected function formatQuote($type, $id, &$quoteData, $format = false) {
             $format = c('Garden.InputFormatter');
         }
 
-        $discussionModel = new DiscussionModel();
         $type = strtolower($type);
         switch ($type) {
             case 'comment':
                 $commentModel = new CommentModel();
                 $data = $commentModel->getID($id);
-                $discussion = $discussionModel->getID(val('DiscussionID', $data));
+                $discussion = $this->discussionModel->getID(val('DiscussionID', $data));
```

Comment:
I think you're fine here, but as a cautionary note (in case you weren't already considering it): you're going from a one-off instance to a persistent instance. This means the state of that instance will persist between usages across the various methods in this class, which can introduce unexpected consequences.